src/main/java/frc/robot/BallChucker9000.java

```diff
@@ -0,0 +1,106 @@
+/*
+     Rotator:
+      Motor - Positive for clockwise, negative for counterclockwise
+      Encoder
+      Limit Switch - False for closed, true for open
+     
+     Flywheel:
+      Motor - Positive will eject the ball
+      Encoder
+     
+      Indexer
+       Motor - Positive will push balls toward the flywheel
+       Piston - True for extended, false for retracted
+
+      - - - - - - - - - - - - - - - - - - - - - - -
+
+      Professionally chucking balls since 2020
+*/
+
+package frc.robot;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.Encoder;
+import edu.wpi.first.wpilibj.Solenoid;
+import edu.wpi.first.wpilibj.VictorSP;
+
+public class BallChucker9000 {
+
+    // ESC declarations
+    private final VictorSP rotatorMotor;
+    private final VictorSP flywheelMotor;
+    private final VictorSP indexerMotor;
+    
+    // Encoder declarations
+    private final Encoder rotatorEncoder;
+    private final Encoder flywheelEncoder;
+
+    // Piston declarations
+    private final Solenoid indexerPiston;
+
+    // Limit switches
+    private final DigitalInput rotatorAtZeroSwitch;
+
+
+    // Class initializer
+    public BallChucker9000(int flywheelMotorPort, int rotatorMotorPort, int indexerMotorPort, int rotatorEncoderChannelA, 
+                            int rotatorEncoderChannelB, int flywheelEncoderChannelA, int flywheelEncoderChannelB, 
+                            int indexerPistonPort, int rotatorAtZeroSwitchPort) {
+    
+        // ESCs
+        flywheelMotor = new VictorSP(flywheelMotorPort);
+        rotatorMotor = new VictorSP(rotatorMotorPort);
+        indexerMotor = new VictorSP(indexerMotorPort);
+
+        // Encoders
+        rotatorEncoder = new Encoder(rotatorEncoderChannelA, rotatorEncoderChannelB);
+        flywheelEncoder = new Encoder(flywheelEncoderChannelA, flywheelEncoderChannelB);
+        rotatorEncoder.setDistancePerPulse(1.0/1024.0);
+        flywheelEncoder.setDistancePerPulse(1.0/1024.0);
+
+        // Piston
+        indexerPiston = new Solenoid(indexerPistonPort);
+
+        //Limit switches
+        rotatorAtZeroSwitch = new DigitalInput(rotatorAtZeroSwitchPort);
+
+    }
+
+    // Setters
+
+    // ESC
+    public void rotatorMotorControl(double power) {
+        rotatorMotor.set(power);
+    }
+
+    public void flywheelMotorControl(double power) {
+        flywheelMotor.set(power);
+    }
+
+    public void indexerMotorControl(double power) {
+        indexerMotor.set(power);
+    }
+
+    // Pistons
+    public void indexerPistonOut(boolean state) {
+        indexerPiston.set(state);
+    }
+
+
+    // Getters
+
+    // Encoder
+    public double getRotatorEncoder() {
+        return rotatorEncoder.getDistance();
+    }
+
+    public double getFlywheelEncoder() {
+        return flywheelEncoder.getRate();
+    }
+
+    // Limit switch 
+    public boolean getRotatorAtZeroSwitch() {
+        return !rotatorAtZeroSwitch.get();
+    }
+
```

Comment:
Schuyler is correct. There should be a "void update" function. It can set the rotatorEncoder to 0 when the switch is pressed. Please include a comment saying that the value 0 can be changed later depending on how where the limit switch is placed.